### ISSUE TITLE
Merge the forks

### ISSUE THREAD

**Issue by nlehuby** [CanalTP/geocoder-tester#4]
Hey,

we need to exterminate the forks and use the same geocoder-tester repo for all our tests on Mimir \o/

![exterminate !](https://media.giphy.com/media/yO8CVuP8X4i1G/giphy.gif)

Some of your tests won't be usable for us and the [geocoders](http://github.com/geocoders) community:
* your Paris tests are too specific (there are a few other Paris in the world more relevant than "Le Touquet-Paris-Plage")
* your tests on NPDC stop areas are great, but "Bourg" and "Terrasse" will never actually work for anyone

I can see 3 solutions:
* re-work on your specific tests so anyone can use them, and recontribute them to the original repo
* dont change your test but rebase regularly to keep this repo up to date
* split the repo to isolate your specific tests, then make some magic so we can launch the tests from both this repo and your specific repo. @antoine-de will give you some tips on that :wink:

**Comment by WilliamFoster** [comment]
Hi, 
we have discussed this a bit with @kinnou02 and the solution : 

> "re-work on your specific tests so anyone can use them, and recontribute them to the original repo"

seems the best for everyone and will not cost us too much to do the modifications. 

The paris tests needs to be changed to something more precise right ?  and that works worldwide ? 
For the stop areas we could discuss on changing the parts that poses problems.
(actually if you can give me some directions to change them i can do it)

**Comment by antoine-de** [comment]
the problem is that this way you won't be able to do tests on the stops in nord-pas-de-calais since only `Kisio` can run those tests because the data are not open :confused: 

I think it's indeed the only tests that might live in another repository, all the others can be adapted and contributed back to geocoder-tester.

As for putting some tests apart of geocoder-tester, it's a bit cumbersome due to the architecture of `pytest`.
The only solution I found to have your tests apart is:
* have another repository with only your tests **+ a copy of [geocoder-tester's conftest](https://github.com/geocoders/geocoder-tester/blob/master/conftest.py)** and to run the test do something like:
`pytest --api-url gnagna <path_to_some_tests_in_geocoder_tester> <path_to_your_tests>`
(pytest load for each test only the 'hierarchy' of conftest.py , so if we want tests in csv or yaml it need to find a conftest.py with the logic to read them)
* patch the conftest.py to add another parameter `--external-directory`.
I tried doing this but since pytest does not publish lots of its internal functions (and it's quite normal), I found myself having to duplicate lots of pytest code to make this work, so I don't think it's a good approach.
However I'm not a pytest expert at all, so maybe someone can find a better way to acheive this.

**Comment by WilliamFoster** [comment]
can't we juste change the coverage and the tests  have them tested in an open data ? it might take some time but would be useful for everyone.

**Comment by antoine-de** [comment]
it's better but mimir would be the only one to be able to run tests on stops I think. Howver if you put those tests in specific directory (a bit like the poi), they might accept the contribution.

You could maybe open an issue on the central repository to ask them this.

**Comment by nlehuby** [comment]
Here are our first results for `Paris` query:

name                     | lat               | lon                | distance | type          
------------------------ | ----------------- | ------------------ | -------- | --------------
Paris                    | 48.8566101        | 2.3514992          | —        | state_district
Paris                    | 48.8566101        | 2.3514992          | —        | city          
Paris-l'Hôpital          | 46.9150284        | 4.633022599999999  | —        | city          
Paris Island Big         | 54.41576792717073 | -7.692538818588079 | —        | city          
Paris Island Little      | 54.41247153846727 | -7.692727710053521 | —        | city          




*About the tests on stop_areas:*
It think it is ok to add them to geocoder-tester, we already have some tests on train stations in Germany : https://github.com/geocoders/geocoder-tester/blob/master/geocoder_tester/world/germany/poi/test_train_stations.csv

On your NPDC tests, if I change the limit to 4 and use the `--loose-compare` parameter, I get `43 failed, 60 passed, 6 skipped`.

We have imported POIs for all Europe, and we have among them rail, subway and tram stations (but no bus stops). 
I also needed to change your `expected_type` to `poi` as we import public transport objects as poi for now. The expected_type `public_transport:stop_area` will indeed need to be changed if we want to contribute back these tests because it is quite Mimir specific.

Here are our detailed results if you want to have a look : https://pastebin.com/TAhMEA9n
We have 
- some tests that need to be changed or discarded (query `Gare de Lyon`, expected `gare de Lyon-Vaise`)
- some that need to be more precise (query `Saint Maur`, `Buisson` or `Campagne`). Maybe add lat/lon or the city name to the query ?
- some where the stop has the same name as another poi around, and we get that poi instead : `Stade Pierre Mauroy` or `Oscar Lambret` (this is also the name of a hospital nearby)
- some that are almost ok but have formatting issues (`Wasquehal - Hôtel de Ville` instead of `Wasquehal Hotel de Ville`)
